src/App.tsx

```diff
@@ -15,49 +16,126 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort();
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort((itemA, itemB) => itemA.length - itemB.length);
+      break;
+
+    default:
+      break;
+  }
```

Comment:
You can use 1 sort and change it condition inside callback
```suggestion
  visibleGoods.sort((itemA, itemB) => {
      switch (coniditon) {
        case 1:
          break;
 
        case 2:
          break;

      default:
        break;
    }
  }
```